apiserver/facades/agent/unitassigner/unitassigner.go

```diff
@@ -6,187 +6,103 @@ package unitassigner
-// for testing.
-type assignerState interface {
-	WatchForUnitAssignment() state.StringsWatcher
-	AssignStagedUnits(allSpaces network.SpaceInfos, ids []string) ([]state.UnitAssignmentResult, error)
-	AssignedMachineId(unit string) (string, error)
-}
-
-// MachineService is the interface that is used to interact with the machine
-// domain.
-type MachineService interface {
-	CreateMachine(context.Context, machine.Name) (machine.UUID, error)
+// API is an empty struct that backs the (not implemented) endpoint methods.
+type API struct {
+	resources facade.Resources
 }
 
-// NetworkService is the interface that is used to interact with the
-// network spaces/subnets.
-type NetworkService interface {
-	// GetAllSpaces returns all spaces for the model.
-	GetAllSpaces(ctx context.Context) (network.SpaceInfos, error)
-}
+// AssignUnits is not implemented. It always returns a list of nil ErrorResults.
+func (a *API) AssignUnits(ctx context.Context, args params.Entities) (params.ErrorResults, error) {
+	results := params.ErrorResults{
+		Results: make([]params.ErrorResult, len(args.Entities)),
+	}
 
-// ApplicationService is the interface that is used to interact with the
-// application domain.
-type StatusService interface {
-	// SetUnitAgentStatus sets the status of the agent of the given unit.
-	SetUnitAgentStatus(ctx context.Context, name unit.Name, status status.StatusInfo) error
+	return results, nil
 }
 
-// API implements the functionality for assigning units to machines.
-type API struct {
-	st             assignerState
-	machineService MachineService
-	networkService NetworkService
-	statusService  StatusService
-	clock          clock.Clock
-	res            facade.Resources
-}
+// WatchUnitAssignments returns a strings watcher that is notified when new unit
+// assignments are added to the db.
+func (a *API) WatchUnitAssignments(ctx context.Context) (params.StringsWatchResult, error) {
+	// Create a simple watcher that sends the empty string as initial event.
+	w := newEmptyStringWatcher()
 
-// AssignUnits assigns the units with the given ids to the correct machine. The
-// error results are returned in the same order as the given entities.
-func (a *API) AssignUnits(ctx context.Context, args params.Entities) (params.ErrorResults, error) {
-	result := params.ErrorResults{}
-
-	// state uses ids, but the API uses Tags, so we have to convert back and
-	// forth (whee!).  The list of ids is (crucially) in the same order as the
-	// list of tags.  This is the same order as the list of errors we return.
-	ids := make([]string, len(args.Entities))
-	for i, e := range args.Entities {
-		tag, err := names.ParseUnitTag(e.Tag)
-		if err != nil {
-			return result, err
-		}
-		ids[i] = tag.Id()
+	changes, ok := <-w.Changes()
+	if !ok {
+		return params.StringsWatchResult{}, watcher.EnsureErr(w)
 	}
+	return params.StringsWatchResult{
+		StringsWatcherId: a.resources.Register(w),
```

Comment:
I believe `resources` is deprecated in favour of `watcherRegistry`

Reply:
Good one, done!